### ISSUE TITLE
`oras tag` should not create referrers tags

### ISSUE THREAD

**Issue by qweeah** [oras-project/oras#1436]
Although tagging a referrer artifact manifest is discouraged, it is still doable. However, re-tagging is implemented as pushing the exact manifest again to the remote server. According to the distribution-spec, if the remote server does not support referrers API, it will go for the referrers tag schema case. However, it is not necessary since the manifest is already there, we are just adding a new tag not really pushing a new manifest.

In summary, tagging a referrers manifest should never go for the tag schema even if the server does not support referrers API since the referrer manifest is already there on the server.

_Originally posted by @shizhMSFT in https://github.com/oras-project/oras/issues/1435#issuecomment-2197136662_

**Comment by qweeah** [comment]
BTW this only happens in registries that haven't implemented [OCI Distribution Spec v1.1.0](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md).

**Comment by qweeah** [comment]
@njucjc Can you help confirm if this has met your expectation described in #1435? You may build from main branch or use `ghcr.io/oras-project/oras:main`. Thanks in adavnce.

**Comment by njucjc** [comment]
> @njucjc Can you help confirm if this has met your expectation described in #1435? You may build from main branch or use `ghcr.io/oras-project/oras:main`. Thanks in adavnce.

@qweeah I already try it out, It works

**Comment by qweeah** [comment]
Closing as it's resolved.
